### PR TITLE
feat: split combine and combineWithAllErrors into sync and async versions

### PR DESCRIPTION
**Pull request by tam-carre**
Resolves https://github.com/supermacro/neverthrow/issues/381

Introduces breaking API changes

This PR removes the `combine` and `combineWithAllErrors` exports and creates `Result.combine`, `Result.combineWithAllErrors`, `ResultAsync.combine`, `ResultAsync.combineWithAllErrors`.

The "Utilities" section of the README has been removed as it would become just references to Result and ResultAsync sections.

Note that this PR makes a slightly different use of `any` from the previous code, but it is equivalent.

### CODE REVIEW

**Comment by supermacro** [comment]
ETA for review + approval is next week

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "neverthrow",
-  "version": "4.3.1",
+  "version": "4.4.2",
```
Comment (supermacro) [inline package.json:3]: Something seems odd here ... the `master` branch is already at 4.4.2, so we shouldn't be seeing a diff here.

**Review by supermacro — CHANGES_REQUESTED** [review]

**Comment by tam-carre** [comment]
Don't know how to clean up the commit history, made a new PR: https://github.com/supermacro/neverthrow/pull/405

**Comment by tam-carre** [comment]
clean pr: https://github.com/supermacro/neverthrow/pull/406

**Pull request closed without merging**